File: src/net/theharrisoncrafter/Commands/Ping.java
package net.theharrisoncrafter.Commands;

import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class Ping implements CommandExecutor{
	public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args){		
		if(sender instanceof Player){
			if(sender.hasPermission("GodCraft.Ping")){
				sender.sendMessage(ChatColor.GREEN + "Pong");
				return true;
			}
			else{
				sender.sendMessage(ChatColor.RED + "You do not have permissions to do that!");
				return true;
			}
		}
		else{
			sender.sendMessage(ChatColor.YELLOW + "Why would you do this?");
			return true;
		}
	}
}

File: src/main/kotlin/co/orre/godcraft/Commands/Ping.kt
package co.orre.godcraft.Commands

import co.orre.godcraft.God
import org.bukkit.ChatColor as CC
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class Ping(val plugin: God) : CommandExecutor {
    override fun onCommand(sender: CommandSender, cmd: Command, commandLabel: String, args: Array<String>): Boolean {
        if (sender is Player) {
            if (sender.hasPermission("godcraft.ping")) {
                sender.sendMessage("${CC.GREEN}Pong!")
                plugin.logDebug("Received a ping from ${sender.name}")
                return true
            } else {
                sender.sendMessage("${CC.RED}You do not have permissions to do that!")
                return true
            }
        } else {
            sender.sendMessage("${CC.YELLOW}Why would you do this?")
            plugin.logDebug("Console is being a smartass")
            return true
        }
    }
}
